### PR TITLE
global update

### PR DESCRIPTION
**Pull request by fanatid**
Didn't see this code before, mostly this PR apply standard

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# browserify-rsa
```
Comment (calvinmetcalf) [inline README.md:1]: why are we renaming the readme ? it was lowercase to make completion from the comand line easier

File: package.json [hunk 2]
```diff
@@ -2,23 +2,32 @@
   "name": "browserify-rsa",
   "version": "4.0.1",
   "description": "RSA for browserify",
+  "keywords": [],
```
Comment (calvinmetcalf) [inline package.json]: we should really only add the keywords array if we're adding some keywords to the project

File: package.json [hunk 3]
```diff
@@ -2,23 +2,32 @@
...
+  "bugs": {
+    "url": "https://github.com/cryptocoinjs/secp256k1-node/issues"
```
Comment (calvinmetcalf) [inline package.json]: wrong url

File: package.json [hunk 4]
```diff
@@ -2,23 +2,32 @@
...
+  },
+  "license": "MIT",
+  "author": "crypto-browserify contributors",
```
Comment (calvinmetcalf) [inline package.json]: this is an optional field so I'd just skip it

File: package.json [hunk 5]
```diff
@@ -2,23 +2,32 @@
...
+  "files": [
+    "index.js"
+  ],
```
Comment (calvinmetcalf) [inline package.json:11]: readme and license should be in here as well

File: README.md (hunk 1 above)
Comment (fanatid) [inline README.md:1]: I never thought about it, but for node packages usually README.md uses, right?

File: package.json (hunk 5 above)
Comment (fanatid) [inline package.json:11]: nope, they included automatically
was investigated here: https://github.com/cryptocoinjs/coininfo/pull/19#commitcomment-16202076

File: index.js [hunk 6]
```diff
@@ -1,40 +1,33 @@
...
-}
-crt.getr = getr;
-function getr(priv) {
-  var len = priv.modulus.byteLength();
-  var r = new bn(randomBytes(len));
-  while (r.cmp(priv.modulus) >=  0 || !r.umod(priv.prime1) || !r.umod(priv.prime2)) {
-    r = new bn(randomBytes(len));
+
+function getr (priv) {
+  for (var r, len = priv.modulus.byteLength(); ;) {
```
Comment (calvinmetcalf) [inline index.js]: why change this to a for loop, if we are using like a while loop?

Comment (fanatid) [inline index.js]: do you think `while` will be more readable?

File: README.md (hunk 1 above)
Comment (calvinmetcalf) [inline README.md:1]: the tendency is to make the special files allcaps, but they are technically case insensitive due to windows file system being case insensitive and most mac file systems being mostly case insensitive.

File: package.json (hunk 3 above)
Comment (fanatid) [inline package.json]: sorry, updated

File: package.json [hunk 7]
```diff
@@ -2,23 +2,30 @@
...
   "main": "index.js",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com:crypto-browserify/browserify-rsa.git"
+  },
   "scripts": {
-    "test": "node test.js | tspec"
+    "lint": "standard",
+    "test": "npm run lint && npm run unit",
```
Comment (calvinmetcalf) [inline package.json:19]: any particular reason for removing tap-spec ?

File: README.md (hunk 1 above)
Comment (fanatid) [inline README.md:1]: it's bad that file systems are insensitive, do you really want to keep lowercase?

**Comment by calvinmetcalf** [comment]
thanks for cleaning this up, I have a few nits and questions but it looks overall helpful

File: index.js (hunk 6 above)
Comment (calvinmetcalf) [inline index.js]: in this context yes I believe a while loop is much more readable then the equivalent for loop considering the for loop is acting like the while loop by having an empty 3 argument

File: README.md (hunk 1 above)
Comment (calvinmetcalf) [inline README.md:1]: eh since license is all caps as well, we should probably make this one all caps as well

File: package.json (hunk 7 above)
Comment (fanatid) [inline package.json:19]: nope, this is yours personal preferences?

File: index.js (hunk 6 above)
Comment (fanatid) [inline index.js]: what's about `do-while`?

File: package.json (hunk 7 above)
Comment (calvinmetcalf) [inline package.json:19]: yeah I find it makes the test output easier to read

File: index.js (hunk 6 above)
Comment (calvinmetcalf) [inline index.js]: that would work

File: package.json (hunk 7 above)
Comment (fanatid) [inline package.json:19]: but what if another person prefer another output that based on TAP output?
I'm not against tap-spec, I just think that output should be raw, in this case anybody can decide what output he want to get.

File: index.js (hunk 6 above)
Comment (fanatid) [inline index.js]: updated

File: index.js [hunk 8]
```diff
@@ -1,40 +1,34 @@
...
-crt.getr = getr;
-function getr(priv) {
-  var len = priv.modulus.byteLength();
-  var r = new bn(randomBytes(len));
-  while (r.cmp(priv.modulus) >=  0 || !r.umod(priv.prime1) || !r.umod(priv.prime2)) {
-    r = new bn(randomBytes(len));
+
+function getr (priv) {
+  var len = priv.modulus.byteLength()
+  while (true) {
```
Comment (calvinmetcalf) [inline index.js]: I thought we were going to do a 

do {} while() loop ?

File: package.json (hunk 7 above)
Comment (calvinmetcalf) [inline package.json:19]: ah that would be the reason, ok thats good then

File: index.js (hunk 8 above)
Comment (fanatid) [inline index.js]: Yes, I wanted use `do-while` here, but current looks better than:

``` js
  var len = priv.modulus.byteLength()
  var r
  do {
    r = new BN(randomBytes(len))
  } while (r.cmp(priv.modulus) >= 0 || !r.umod(priv.prime1) || !r.umod(priv.prime2))
  return r
```

agree?

Comment (calvinmetcalf) [inline index.js]: wouldn't the equivalent be 

``` js
  var len = priv.modulus.byteLength()
  do {
    var r = new BN(randomBytes(len))
  } while (r.cmp(priv.modulus) >= 0 || !r.umod(priv.prime1) || !r.umod(priv.prime2))
  return r
```

Comment (fanatid) [inline index.js]: you're right, it will work, but I don't like idea return variable that was defined in nested block scope

Comment (calvinmetcalf) [inline index.js]: but that's what your code does currently

Comment (fanatid) [inline index.js]: sorry, I mean return variable _from current scope_ that was defined in nested scope
right now variable defined and returned from one scope

Comment (fanatid) [inline index.js]: for me (personal preference):

not ok:

```
function f () {
 if (1) { var x = 2 }
 return x
}
```

ok:

```
function f () {
 if (1) { var x = 1; return x }
}
```

File: index.js [hunk 9]
```diff
@@ -1,40 +1,34 @@
...
+    var r = new BN(randomBytes(len))
+    if (r.cmp(priv.modulus) < 0 && r.umod(priv.prime1) && r.umod(priv.prime2)) return r
   }
-  return r;
```
Comment (calvinmetcalf) [inline index.js:39]: woops I thought this was still in

Comment (fanatid) [inline index.js:39]: it's ok :) so, what you're prefer here: current `while` or `do-while` ?

File: index.js (hunk 8 above)
Comment (calvinmetcalf) [inline index.js]: yeah I was mis reading the change, I thought you were still returning r, anywho I don't think the while true is much of an improvement, we should probably either ditch this change for now and not include it in this particular pull.

Comment (fanatid) [inline index.js]: okay, updated to old code
can you make change without pull after merge?

**Comment by fanatid** [comment]
@calvinmetcalf Thank you!
Any plans about publishing changes? At least new code uses `toArrayLike` from `bn.js`.

**Comment by dcousens** [comment]
@calvinmetcalf in a `.inputrc`:

``` bash
# tab complete should ignore case
set completion-ignore-case on
set completion-map-case on
```

**Pull request merged**